### PR TITLE
fix(browser): unflake TestPageOnResponse nil panic

### LINKED ISSUE #5124: `TestPageOnResponse` is flaky
The test randomly fails with a `nil` pointer panic:

```
--- FAIL: TestPageOnResponse (0.70s)
    page_test.go:2556: 
        	Error Trace:	/home/runner/work/k6/k6/internal/js/modules/k6/browser/tests/page_test.go:2556
        	Error:      	Received unexpected error:
        	            	error building browser on IterStart: canceled at <eval>:1:1(0)
        	Test:       	TestPageOnResponse
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1a393e3]

goroutine 493 [running]:
testing.tRunner.func1.2({0x1e015e0, 0x2f373b0})
	/home/runner/sdk/gotip/src/testing/testing.go:1874 +0x419
testing.tRunner.func1()
	/home/runner/sdk/gotip/src/testing/testing.go:1877 +0x67f
panic({0x1e015e0?, 0x2f373b0?})
	/home/runner/sdk/gotip/src/runtime/panic.go:858 +0x13a
go.k6.io/k6/internal/js/modules/k6/browser/k6ext/k6test.ToPromise({0x2303030, 0xc0002cddc0}, {0x0, 0x0})
	/home/runner/work/k6/k6/internal/js/modules/k6/browser/k6ext/k6test/vu.go:167 +0x63
go.k6.io/k6/internal/js/modules/k6/browser/tests.TestPageOnResponse(0xc0002cddc0)
	/home/runner/work/k6/k6/internal/js/modules/k6/browser/tests/page_test.go:2558 +0x529
testing.tRunner(0xc0002cddc0, 0x20995a0)
	/home/runner/sdk/gotip/src/testing/testing.go:1936 +0x21d
created by testing.(*T).Run in goroutine 1
	/home/runner/sdk/gotip/src/testing/testing.go:1999 +0x9d3
```

- k6 version: v1.2.3
- #4296 
- #4234

### PR DESCRIPTION
**Pull request by alirezaedalat**
## Summary
- Stop launching an unused Go-level Chromium in `TestPageOnResponse` via `withSkipLaunch()`; the test only needs the HTTP server and the managed browser from `StartIteration`. Double launches contributed to flaky `error building browser on IterStart: canceled` under CI load.
- Harden `k6test.ToPromise` against a nil `sobek.Value` so interrupted event loops fail the test cleanly instead of panicking on `Export()` (the panic in #5124).
- Switch the test to `RunPromise` and add a regression unit test for the nil-value path.

Fixes #5124

## Test plan
- [x] `go test -run TestToPromiseNilValue ./internal/js/modules/k6/browser/k6ext/k6test/`
- [x] `go test -run '^TestPageOnResponse$' ./internal/js/modules/k6/browser/tests/`
- [x] `go test -count=10 -parallel 4 -run '^TestPageOnResponse$' ./internal/js/modules/k6/browser/tests/`
- [ ] CI browser tests on this PR

### CODE REVIEW

**Comment by inancgumus** [comment]
@alirezaedalat FYI: https://github.com/grafana/k6/pull/6258#issuecomment-5209474358

> "This repository requires all commits to be signed."

**Comment by alirezaedalat** [comment]
Thanks @inancgumus — signed the commit. CLA and signed-commits should be green now.

File: internal/js/modules/k6/browser/tests/test_browser.go [hunk 1]
```diff
@@ -82,6 +88,13 @@ func newTestBrowser(tb testing.TB, opts ...func(*testBrowser)) *testBrowser {
 	tbr.isBrowserTypeInitialized = true // some option require the browser type to be initialized.
 	tbr.applyOptions(opts...)           // apply post-init stage options.
 
+	if tbr.skipLaunch {
+		// Keep a usable context for helpers that call testBrowser.context()
+		// even when no Go-level browser is launched.
+		tbr.ctx = tbr.vu.Context()
+		return tbr
```
Comment (inancgumus) [inline internal/js/modules/k6/browser/tests/test_browser.go:95]: Does this early return leave the embedded `*common.Browser` nil, so the next test that calls `tb.NewPage` panics? A `t.Fatalf` in `NewPage` would read like the one `url()` gives when the HTTP server is off.

File: internal/js/modules/k6/browser/tests/page_test.go [hunk 2]
```diff
@@ -2569,8 +2569,12 @@ type response struct {
 func TestPageOnResponse(t *testing.T) {
 	t.Parallel()
 
-	// Start and setup a webserver to test the page.on('request') handler.
-	tb := newTestBrowser(t, withHTTPServer())
+	// Skip the Go-level Chromium launch: this test only needs the HTTP server
+	// and drives the browser through the JS module after StartIteration.
+	// Launching both (newTestBrowser + IterStart) doubles Chromium processes
+	// and has caused flaky "error building browser on IterStart: canceled"
```
Comment (inancgumus) [inline internal/js/modules/k6/browser/tests/page_test.go:2575]: Was the double launch measured as the cause here? #5492 hits the same `IterStart` abort with only one browser per VU, so does dropping it close #5124 or make it rarer?

File: internal/js/modules/k6/browser/tests/test_browser.go [hunk 3]
```diff
@@ -70,6 +72,10 @@ type testBrowser struct {
 //   - withLogCache: enables the log cache.
 //   - withSamples: provides a channel to receive the browser metrics.
 //   - withSkipClose: skips closing the browser when the test finishes.
+//   - withSkipLaunch: skips launching the Go-level browser. Use this when the
+//     test only needs the VU + HTTP server and will build a managed browser via
+//     StartIteration (e.g. scripts that call browser.newContext()). Launching
+//     both browsers doubles Chromium processes and contributes to CI flakiness.
```
Comment (mstoykov) [inline internal/js/modules/k6/browser/tests/test_browser.go:78]: Go-level browser sounds strange to me. Can we go with "local browser" or something like this, as that usually the thing that is meant in this cases

File: internal/js/modules/k6/browser/k6ext/k6test/vu_test.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		defer func() { _ = recover() }()
+		ToPromise(tb, nil)
+	}()
+	<-done
+
+	require.True(t, tb.failed, "ToPromise(nil) should fail the test without panicking")
+}
```
Comment (mstoykov) [inline internal/js/modules/k6/browser/k6ext/k6test/vu_test.go:43]: I find this test and file to not really be needed

**Review by mstoykov — APPROVED** [review]
LGTM in general. I do not think we need to test the test helper and add additional compolexity to the tests.

And I also find hte "go-level" wording ... strange. Apart  from that it seems okay to me.

I am approving this in case we decided to merge it as is and fix later

**Review by inancgumus — APPROVED** [review]
Thanks for your contribution!
